### PR TITLE
[Don't Merge]Add configuration item to support configuring replication adapters

### LINKED ISSUE #9342: Make the supported of replication adapters configurable
Provide a configuration option to config which adapters are available when do the replication

### PR DESCRIPTION
**Pull request by ywk253100**
Fixes #9342, add configuration item to support configuring replication adapters

Signed-off-by: Wenkai Yin <yinw@vmware.com>

### CODE REVIEW

File: make/photon/prepare/utils/configs.py [hunk 1]
```diff
@@ -310,4 +310,7 @@ def parse_yaml_config(config_file_path, with_notary, with_clair, with_chartmuseu
     # UAA configs
     config_dict['uaa'] = configs.get('uaa') or {}
 
+    # replication adapters
+    config_dict['replication_adapters'] = configs.get("replication_adapters") or ""
```
Comment (bitsf) [inline make/photon/prepare/utils/configs.py]: config_dict['replication_adapters'] = configs.get("replication_adapters", "")

File: src/replication/adapter/adapter.go [hunk 2]
```diff
@@ -140,23 +141,42 @@ func RegisterFactory(t model.RegistryType, factory Factory) error {
 // GetFactory gets the adapter factory by the specified name
 func GetFactory(t model.RegistryType) (Factory, error) {
 	factory, exist := registry[t]
-	if !exist {
+	if !exist || !isAdapterSupported(t) {
```
Comment (stonezdj) [inline src/replication/adapter/adapter.go]: break it into two if will make it easy to read

File: make/harbor.yml [hunk 3]
```diff
@@ -161,3 +161,6 @@ proxy:
     - core
     - jobservice
     - clair
+
+# Comma-separated adapters for replication. All adapters are supported if not specify 
+# replication_adapters:
```
Comment (stonezdj) [inline make/harbor.yml]: Provide a full list sample in the comment
#replication_adapters: docker-registry, docker-hub- docker-registry,docker-hub,harbor,helm-hub,google-gcr,aws-ecr

Comment (reasonerjt) [reply make/harbor.yml]: +1 I'd suggest we provide a full list of yaml and comment the ones we don't support by default, so user knows how to enable them on their own risk.

**Comment by reasonerjt** [comment]
Closing this PR as it is determined to be moved out of 1.10.0

**Pull request closed without merging**